diff -x '*.pyc' -ru fabfile-3.0-2723.orig/tasks/helpers.py fabfile-3.0-2723/tasks/helpers.py
--- fabfile-3.0-2723.orig/tasks/helpers.py	2016-02-29 18:44:50.000000000 -0800
+++ fabfile-3.0-2723/tasks/helpers.py	2016-03-05 11:12:19.572433000 -0800
@@ -1313,7 +1313,8 @@
         exit(1)
     else:
         print "\nINFO: \n\tOpenstack and cfgm nodes are same, No need for contrail_internal_vip to be specified in testbed.py."
-    execute('add_openstack_reserverd_ports')
+    if env.roledefs['openstack']:
+        execute('add_openstack_reserverd_ports')
 
 def role_to_ip_dict(role=None):
     role_to_ip_dict = {}
diff -x '*.pyc' -ru fabfile-3.0-2723.orig/tasks/install.py fabfile-3.0-2723/tasks/install.py
--- fabfile-3.0-2723.orig/tasks/install.py	2016-02-29 18:44:50.000000000 -0800
+++ fabfile-3.0-2723/tasks/install.py	2016-03-05 11:12:19.600433000 -0800
@@ -579,16 +579,14 @@
 @roles('build')
 def create_install_repo_without_openstack(*tgzs, **kwargs):
     """Creates contrail install repo in all nodes excluding openstack node."""
-    if len(tgzs) == 0:
-        cmd = 'create_install_repo_node'
-    else:
-        cmd = 'create_install_repo_from_tgz_node'
-
     for host_string in env.roledefs['all']:
         if host_string in env.roledefs['openstack']:
             continue
         with settings(host_string=host_string):
-            execute(cmd, host_string, *tgzs, **kwargs)
+            if len(tgzs) == 0:
+                execute('create_install_repo_node', host_string)
+            else:
+                execute('create_install_repo_node', host_string, *tgzs, **kwargs)
 
 @task
 def create_install_repo_from_tgz_node(host_string, *tgzs, **kwargs):
diff -x '*.pyc' -ru fabfile-3.0-2723.orig/tasks/provision.py fabfile-3.0-2723/tasks/provision.py
--- fabfile-3.0-2723.orig/tasks/provision.py	2016-02-29 18:44:50.000000000 -0800
+++ fabfile-3.0-2723/tasks/provision.py	2016-03-05 11:12:19.576433000 -0800
@@ -2466,7 +2466,8 @@
 
 @roles('build')
 @task
-def setup_without_openstack(manage_nova_compute='yes', reboot='True'):
+def setup_without_openstack(manage_nova_compute='yes', config_nova = 'yes',
+        reboot='True'):
     """Provisions required contrail packages in all nodes as per the role definition except the openstack.
        User has to provision the openstack node with their custom openstack pakckages.
        If manage_nova_compute = no; Only vrouter services is provisioned, nova-compute will be skipped in the compute node.
@@ -2485,7 +2486,7 @@
     execute('verify_collector')
     execute('setup_webui')
     execute('verify_webui')
-    execute('setup_vrouter', manage_nova_compute)
+    execute('setup_vrouter', manage_nova_compute, config_nova)
     execute('prov_config')
     execute('prov_database')
     execute('prov_analytics')
diff -x '*.pyc' -ru fabfile-3.0-2723.orig/utils/install.py fabfile-3.0-2723/utils/install.py
--- fabfile-3.0-2723.orig/utils/install.py	2016-02-29 18:44:50.000000000 -0800
+++ fabfile-3.0-2723/utils/install.py	2016-03-05 13:39:17.220433000 -0800
@@ -77,7 +77,7 @@
                'contrail-vrouter-common']
 	if get_orchestrator() is 'openstack':
             pkgs.append('contrail-nova-vif')
-        else:
+        elif get_orchestrator() is 'vcenter':
             pkgs.append('contrail-vmware-vrouter')
     # Append lbaas dependent packages if haproxy is enabled..
     if getattr(testbed, 'haproxy', False):
